tap_core/src/tap_receipt/received_receipt.rs

```diff
@@ -14,63 +14,136 @@
 //! their progress through various checks and stages of inclusion in RAV requests and received RAVs.
 
 use serde::{Deserialize, Serialize};
-use strum_macros::{Display, EnumString};
 
-use super::{
-    receipt_auditor::ReceiptAuditor, Receipt, ReceiptCheck, ReceiptCheckResults, ReceiptError,
-};
+use super::{receipt_auditor::ReceiptAuditor, Receipt, ReceiptCheck, ReceiptCheckResults};
 use crate::{
-    adapters::{escrow_adapter::EscrowAdapter, receipt_checks_adapter::ReceiptChecksAdapter},
+    adapters::{
+        escrow_adapter::EscrowAdapter, receipt_checks_adapter::ReceiptChecksAdapter,
+        receipt_storage_adapter::StoredReceipt,
+    },
     eip_712_signed_message::EIP712SignedMessage,
     Error, Result,
 };
 
-#[derive(Eq, PartialEq, Debug, Clone, EnumString, Display, Serialize, Deserialize)]
-/// State of the contained receipt
-pub enum ReceiptState {
-    /// Initial state, received with no checks started
-    Received,
-    /// Checking in progress, no errors found
-    Checking,
-    /// Checks completed with at least one check resulting in an error
-    Failed,
-    /// Checks completed with all passed, awaiting escrow check and reserve
-    AwaitingReserveEscrow,
-    /// All checks completed with no errors found, escrow is reserved if requested by user
-    Accepted,
-    /// Receipt was added to a RAV request
-    IncludedInRAVRequest,
-    /// Receipt was included in received RAV
-    Complete,
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct Checking {
+    /// A list of checks to be completed for the receipt, along with their current result
+    pub(crate) checks: ReceiptCheckResults,
+}
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct Failed;
```

Comment:
Can we keep the `ReceiptCheckResults` in `Failed` so that the user can easily understand what went wrong? I suppose it's going to be especially useful while we're doing integration/testnet tests.